packages/router/src/lib/route-config.ts

```diff
@@ -21,6 +23,44 @@ export function toRouteConfig(routeMeta: RouteMeta | undefined): RouteConfig {
     };
   }
 
+  if (!routeConfig) {
+    routeConfig = {};
+  }
+
+  routeConfig.resolve = {
+    ...routeConfig.resolve,
+    load: async (route) => {
+      const { queryParams, fragment: hash, params } = route;
+      const routeConfig = route.routeConfig as Route & {
+        meta: { endpoint: string; endpointKey: string };
+      };
+
+      if (PAGE_ENDPOINTS[routeConfig.meta.endpointKey]) {
+        const http = inject(HttpClient);
+        const url = new URL('', import.meta.env['VITE_ANALOG_PUBLIC_BASE_URL']);
```

Comment:
We should note somewhere that you will need to add this to your `.env` file or as an environment variable when running the application :) really good idea to introduce this though and I think this could be super helpful for many features

Reply:
Agreed! Will add this with the docs.